src/utils/saml.ts

```diff
@@ -0,0 +1,21 @@
+import { DOMParser } from "xmldom";
+
+export interface ISamlResponse {
+  RelayState: string;
+  SAMLResponse: string;
+}
+
+export const getSamlIssuer = (body: ISamlResponse): string | undefined => {
+  const decodedResponse =
+    body && body.SAMLResponse
+      ? Buffer.from(body.SAMLResponse, "base64").toString("utf8")
+      : "";
+  const domParser = new DOMParser().parseFromString(decodedResponse);
+  if (domParser === undefined) {
+    return undefined;
+  }
+  const samlIssuerTAG = "saml:Issuer";
+  const samlIssuerElements = domParser.getElementsByTagName(samlIssuerTAG);
+  const issuers = Array.from(samlIssuerElements).map(_ => _.textContent);
+  return issuers[0] ? issuers[0] : undefined;
```

Comment:
isn't this code equivalent to `return issuers[0];` ?

Reply:
fixed in d098c6c